pkg/xds/server/callbacks/dataplane_status_sink.go

```diff
@@ -154,8 +155,23 @@ func (s *dataplaneInsightStore) Upsert(dataplaneType core_model.ResourceType, da
 			if err := insight.Spec.UpdateSubscription(subscription); err != nil {
 				return err
 			}
-			if secretsInfo != nil && (insight.Spec.MTLS == nil || insight.Spec.MTLS.CertificateExpirationTime.AsTime() != secretsInfo.Expiration) {
-				return insight.Spec.UpdateCert(secretsInfo.Generation, secretsInfo.Expiration)
+			if secretsInfo == nil { // it means mTLS was disabled, we need to clear stats
+				insight.Spec.MTLS = nil
+			} else if insight.Spec.MTLS == nil || insight.Spec.MTLS.CertificateExpirationTime.AsTime() != secretsInfo.Expiration {
+				if err := insight.Spec.UpdateCert(secretsInfo.Generation, secretsInfo.Expiration, secretsInfo.MTLS.EnabledBackend, nil); err != nil {
+					return err
+				}
+			}
+
+			if secretsInfo == nil { // it means mTLS was disabled, we need to clear stats
+				insight.Spec.MTLS = nil
+			} else if insight.Spec.MTLS == nil ||
+				insight.Spec.MTLS.CertificateExpirationTime.AsTime() != secretsInfo.Expiration ||
+				insight.Spec.MTLS.IssuedBackend != secretsInfo.IssuedBackend ||
+				!reflect.DeepEqual(insight.Spec.MTLS.SupportedBackends, secretsInfo.SupportedBackends) {
+				if err := insight.Spec.UpdateCert(secretsInfo.Generation, secretsInfo.Expiration, secretsInfo.IssuedBackend, secretsInfo.SupportedBackends); err != nil {
+					return err
+				}
```

Comment:
Hmm copy pasta I guess?

Reply:
_how did this happen..._

thanks for noticing.